aiverify-foundation/moonshot-ui
# Feature/total prompts run

## Description

PR includes

- displaying of number of datasets under each cookbook in the run benchmark user flow
- revamp the benchmark run form "run smaller set of prompts" textbox
- run all by toggling a toggle switch
- run smaller set textbox is changed to number type field
- display number of prompts that will be run
- added validation for 0 for that input
- add toggle switch reusable component
- refactor benchmark run form (from formik to server action)
- move the clearing of selected cookbooks and endpoints from the end of the flow to the benchmark run status screen

## Motivation and Context

Basically, a better UI/UX for the 'run smaller set' or 'run all' prompts in the benchmark run form

## Type of Change


 - feat: A new feature










## How to Test

This is dependent on web-api PR [https://github.com/aiverify-foundation/moonshot/pull/337](https://github.com/aiverify-foundation/moonshot/pull/337)

1. Go into the benchmark user flow from home page top banner - "Get Started"
2. Select cookbooks to test, click on next arrow icon
3. On the cookbooks recommendation screen, click "these cookbooks"
4. Verify Number of datasets are displayed on each cookbook card
5. Click on 'About' link on cookbook card
6. Verify Total number of datasets is displayed. Verify number of datasets is displayed under each recipe
7. Close the "About cookbook" screen, then click next arrow icon
8. On the Run form, verify total prompts and number of prompts that will be run (select Run All / or enter no. of prompts per recipe)
9. Verify '0' validation on number of prompts textbox
10. Click Run button. Verify smaller set run / run all.

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
